cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetTest.java

```diff
@@ -547,14 +554,131 @@ public void apply() throws Exception {
   }
 
   @Test
+  public void testRollbackOnTransactionAbort() throws Exception {
+    PartitionedFileSet pfs = dsFrameworkUtil.getInstance(pfsInstance);
+    TransactionContext txContext = new TransactionContext(txClient, (TransactionAware) pfs);
+
+    txContext.start();
+
+    PartitionOutput output = pfs.getPartitionOutput(PARTITION_KEY);
+    Location outputLocation = output.getLocation().append("file");
+    Assert.assertFalse(outputLocation.exists());
+
+    // this will create the file
+    outputLocation.getOutputStream().close();
+    Assert.assertTrue(outputLocation.exists());
+
+    output.addPartition();
+    Assert.assertNotNull(pfs.getPartition(PARTITION_KEY));
+    Assert.assertTrue(pfs.getPartition(PARTITION_KEY).getLocation().exists());
+
+    txContext.abort();
+
+    // because the previous transaction aborted, the partition as well as the file will not exist
+    txContext.start();
+    Assert.assertNull(pfs.getPartition(PARTITION_KEY));
+    Assert.assertFalse(outputLocation.exists());
+    txContext.finish();
+  }
+
+  @Test
+  public void testRollbackOnJobFailure() throws Exception {
+    Map<String, String> args = new HashMap<>();
+    FileSetArguments.setOutputPath(args, "custom/output/path");
+    PartitionedFileSetArguments.setOutputPartitionKey(args, PARTITION_KEY);
+    PartitionedFileSet pfs = dsFrameworkUtil.getInstance(pfsInstance, args);
+    TransactionContext txContext = new TransactionContext(txClient, (TransactionAware) pfs);
+
+    txContext.start();
+
+    Location outputLocation = pfs.getEmbeddedFileSet().getOutputLocation();
+    Assert.assertFalse(outputLocation.exists());
+
+    // this will create the file
+    outputLocation.getOutputStream().close();
+    Assert.assertTrue(outputLocation.exists());
+
+    ((PartitionedFileSetDataset) pfs).onFailure();
+    txContext.abort();
+
+    // because the previous transaction aborted, the partition as well as the file will not exist
+    txContext.start();
+    Assert.assertNull(pfs.getPartition(PARTITION_KEY));
+    Assert.assertFalse(outputLocation.exists());
+    txContext.finish();
+  }
+
+  @Test
+  public void testRollbackOfPartitionDelete() throws Exception {
+    PartitionedFileSet pfs = dsFrameworkUtil.getInstance(pfsInstance);
```

Comment:
that did work before this change, but now there's extra behavior during rollback of that transaction:
the files will be put back into the original location and then deleted. So will add a test case to test that.